Week1/homework/app.js

```diff
@@ -1,11 +1,154 @@
 'use strict';
 
 {
-  const bookTitles = [
+  /*
+  const myBookTitles = [
     // Replace with your own book titles
-    'harry_potter_chamber_secrets',
+    'Siddhartha',
+    'one_hundred_years_of_solitude',
+    'the_metamorphosis',
+    'hamlet',
+    'heidi',
+    'frankenstein',
+    'lolita',
+    'ulysses',
+    'deception',
+    'scandal',
   ];
 
   // Replace with your own code
-  console.log(bookTitles);
+ 
+  function generateList() {
+    const ul = document.createElement('ul');
+
+    for (let i = 0; i < myBookTitles.length; i++) {
+      const book = myBookTitles[i];
+      const listItem = document.createElement('li');
+      listItem.innerHTML = book;
+      ul.appendChild(listItem);
+    }
+    return ul;
+  }
+  document.body.append(generateList());
+*/
+
+  const myBooks = {
+    siddhartha: {
+      title: 'Siddhartha',
+      language: 'English',
+      author: 'Hermann Hesse',
+    },
+    cities_of_salt: {
+      title: 'Cities of Salt',
+      language: 'Arabic',
+      author: 'Abdelrahman Munif',
+    },
+    the_metamorphosis: {
+      title: 'The Metamorphosis',
+      language: 'German',
+      author: 'Franz Kafka',
+    },
+    hamlet: {
+      title: 'Hamlet',
+      language: 'English',
+      author: 'William Shakespeare',
+    },
+    heidi: {
+      title: 'Heidi',
+      language: 'English',
+      author: 'Johanna Spyri',
+    },
+    frankenstein: {
+      title: 'Frankenstein',
+      language: 'English',
+      author: 'Mary Shelley',
+    },
+    lolita: {
+      title: 'Lolita',
+      language: 'English',
+      author: 'Vladimir Nabokov',
+    },
+    ulysses: {
+      title: 'Ulysses',
+      language: 'English',
+      author: 'James Joyce',
+    },
+    my_name_is_red: {
+      title: 'My name is red',
+      language: 'Turkish',
+      author: 'Orhan Pamuk',
+    },
+    sarmada: {
+      title: 'sarmada',
+      language: 'Arabic',
+      author: 'Fadi Azzam',
+    },
+  };
+
+  function booksInfo() {
+    const ul = document.createElement('ul');
+
+    // Use "for in" loop with objects
+    for (const book in myBooks) {
```

Comment:
The instructions asked you to loop over the book titles in the `myBookTitles` array. And use each title to access the correct data from the `myBooks` object.
This is because the order of keys in an object is not guaranteed, while the order of items in an array is guaranteed. See: https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order
So in the way that you build it the page could show the books in a different order on different computers.

Please change the `booksInfo` function so it loops over the `myBookTitles` array.

Reply:
Thank you for your feedback. I'm working on it